### PR TITLE
cli: verbosity

### PR DESCRIPTION
**Pull request by oliver-sanders**
Addresses: #2972 (point 9)
Superseedes: #4181

* Unify, `--verbose`, `--quiet`, `--debug` and make them global.
* Format exceptions nicely unless verbosity > 1 (`--debug` or `-v -v`) in which case write the full traceback.

Side effects:

* The `CYLC_VERBOSE` env var is now exported in job scripts (can change if we don't like this).

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.py` and
  `conda-environment.yml`.
- [x] Appropriate tests are included (unit and/or functional).
- [x] Appropriate change log entry included.
- [x] No documentation update required.

### CODE REVIEW

File: cylc/flow/option_parsers.py [hunk 1]
```diff
@@ -130,16 +203,26 @@ def add_std_option(self, *args, **kwargs):
...
+        )
         self.add_std_option(
             "-v", "--verbose",
-            help="Verbose output mode.",
-            action="store_true", dest="verbose",
-            default=(os.getenv("CYLC_VERBOSE", "false").lower() == "true"))
+            help="Increase verbosity.",
+            dest='verbosity',
+            action='count',
+            default=env_to_verbosity(os.environ)
```
Comment (MetRonnie) [inline cylc/flow/option_parsers.py:217]: Note to self: Does this default work?

File: cylc/flow/parsec/example/test.py [hunk 2]
```diff
@@ -29,7 +29,7 @@
 sys.path.append(
     os.path.join(os.path.dirname(os.path.abspath(__file__)), '../..'))
 
-cylc.flow.flags.verbose = True
+cylc.flow.flags.verbosity = 1
 
 
 class Testcfg(ParsecConfig):
```
Comment (MetRonnie) [inline cylc/flow/parsec/example/test.py:35]: What is this file actually for?

File: cylc/flow/scripts/validate.py [hunk 3]
```diff
@@ -93,7 +93,7 @@ def main(_, options, reg):
     profiler = Profiler(None, options.profile_mode)
     profiler.start()
 
-    if not cylc.flow.flags.debug:
+    if not cylc.flow.flags.verbosity > 1:
```
Comment (MetRonnie) [inline cylc/flow/scripts/validate.py:96]:
```suggestion
    if cylc.flow.flags.verbosity < 2:
```

File: cylc/flow/option_parsers.py (hunk 1 above)
Comment (MetRonnie) [reply cylc/flow/option_parsers.py:217]: Yes

**Review by MetRonnie — APPROVED** [review]

File: cylc/flow/parsec/example/test.py (hunk 2 above)
Comment (oliver-sanders) [reply cylc/flow/parsec/example/test.py:35]: No idea whatsoever, and that is definitely not the right way to do it (will change the value for all tests).

File: cylc/flow/scripts/validate.py (hunk 3 above)
Comment (oliver-sanders) [reply cylc/flow/scripts/validate.py]: Good spot.

File: cylc/flow/parsec/example/test.py (hunk 2 above)
Comment (MetRonnie) [reply cylc/flow/parsec/example/test.py:35]: @hjoliver Can we get rid of this file?

Comment (hjoliver) [reply cylc/flow/parsec/example/test.py:35]: Lost in the mists of time, probably some built-in test during original parsec development Yes, delete it!

File: cylc/flow/scripts/cylc.py [hunk 4]
```diff
@@ -49,16 +48,12 @@ def get_version(long=False):
     return version
 
 
-DESC = '''
-Cylc ("silk") orchestrates complex cycling (and non-cycling) workflows.
-'''
-
 USAGE = f"""{cylc_header()}
-{centered(DESC)}
```
Comment (hjoliver) [inline cylc/flow/scripts/cylc.py:57]: (good move :+1: )

File: tests/unit/test_terminal.py [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+    stderr,
+    monkeypatch,
+    capsys
+):
+    """Test that the CLI formats exceptions appropriately.
+
+    The idea here is that "known" errors (those which subclass CylcError or
+    ParsecError) should be formatted nicely (as opposed to dumping the
+    full traceback to stderr) in interactive mode. This behaviour can be
+    overridden using --debug mode. In non-interactive mode we allways print the
```
Comment (hjoliver) [inline tests/unit/test_terminal.py:138]:
```suggestion
    overridden using --debug mode. In non-interactive mode we always print the
```

File: cylc/flow/scheduler_cli.py [hunk 6]
```diff
@@ -43,6 +44,8 @@
 
 PLAY_DOC = r"""cylc play [OPTIONS] ARGS
 
+Start, restart or resume a workflow.
+
 Start running a workflow, or restart a stopped workflow from its previous
 state, or resume a paused workflow.
```
Comment (hjoliver) [inline cylc/flow/scheduler_cli.py:50]: Agreed
```suggestion
```

**Review by hjoliver — APPROVED** [review]
Nice :+1:

File: cylc/flow/parsec/example/test.py (hunk 2 above)
Comment (hjoliver) [reply cylc/flow/parsec/example/test.py:35]: (I'll push a file deletion patch to this branch, and then merge...)

**Pull request merged**